crates/cranelift/src/func_environ.rs

```diff
@@ -491,6 +497,22 @@ mod typed_continuation_helpers {
             builder.ins().load(I32, mem_flags, self.address, offset)
         }
 
+        /// Sets the value of the `state` field of the continuation object,
+        pub fn set_state(
+            &self,
+            builder: &mut FunctionBuilder,
+            state: wasmtime_continuations::State,
+        ) {
+            let mem_flags = ir::MemFlags::trusted();
+            let offset = wasmtime_continuations::offsets::continuation_object::STATE;
+
+            // Let's make sure that we still represent the State enum as i32.
+            debug_assert!(mem::size_of::<wasmtime_continuations::State>() == mem::size_of::<i32>());
+
+            let v = builder.ins().iconst(I32, state.discriminant() as i64);
```

Comment:
Does the interface of `iconst` mandate that the second argument is `i64` regardless of the first argument?

Reply:
yes